CPP/DesktopToastsCppApp/DesktopNotificationManagerCompat.h

```diff
@@ -0,0 +1,122 @@
+// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED �AS IS�, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+#pragma once
+#include <rpc.h>
+#include <Windows.h>
+#include <windows.ui.notifications.h>
+#include <wrl.h>
+#define TOAST_ACTIVATED_LAUNCH_ARG "-ToastActivated"
+
+using namespace ABI::Windows::UI::Notifications;
+
+class DesktopNotificationHistoryCompat;
+
+class DesktopNotificationManagerCompat
```

Comment:
Why do you want a static only class? The downside here is static won't work well with impersonation. Probably not something real people outside Microsoft deal with very often but...

Reply:
Huh, do you think there's any significant need for impersonation? Another option if it's rare is that those who use impersonation could always edit the library code themselves in their own copy. I personally know nothing about impersonation so not sure whether it's a common requirement.

Reply:
Significant need? Absolutely not. If you prefer this calling pattern then you should change DNMC to be a namespace rather than a class.

Reply:
Ah interesting, didn't even know you can declare methods that aren't part of a static class (coming from C#), but thinking about it now that makes sense! I'll update that code to use that pattern, since sounds like it's preferred by C++ folks.